tests/unit/customizations/test_overridesslcommonname.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from awscli.testutils import mock, create_clidriver
+from awscli.customizations.overridesslcommonname import (
+    update_endpoint_url,
+    ssl_common_names,
+)
+
+import pytest
+
+
+test_cases = [
+    (service, region)
+    for service, regions in ssl_common_names.items()
+    for region in regions
+]
+
+
+def _setup_test():
+    parsed_globals = mock.Mock()
+    parsed_globals.endpoint_url = None
+    parsed_globals.region = None
+    driver = create_clidriver()
+    return parsed_globals, driver.session
+
+
+@pytest.mark.parametrize("service,region", test_cases)
+def test_update_endpoint_url(service, region):
+    parsed_globals, session = _setup_test()
+    parsed_globals.command = service
+    session.set_config_variable("region", region)
+    update_endpoint_url(session, parsed_globals)
+    assert parsed_globals.endpoint_url == f"https://{ssl_common_names[service][region]}"
+
+
+@pytest.mark.parametrize("service,region", test_cases)
+def test_url_modified_from_event(service, region):
+    parsed_globals, session = _setup_test()
+    parsed_globals.command = service
+    session.set_config_variable("region", region)
+    session.emit(
+        "before-building-argument-table-parser.sqs",
+        args=[],
+        session=session,
+        argument_table=mock.Mock(),
+        parsed_globals=parsed_globals,
+    )
+    assert parsed_globals.endpoint_url == f"https://{ssl_common_names[service][region]}"
+
+
+def test_dont_modify_provided_url():
+    parsed_globals, session = _setup_test()
+    parsed_globals.endpoint_url = "https://test.com"
+    parsed_globals.command = "sqs"
```

Comment:
Why are we setting the command for this one but not the others?

Reply:
the command is set for the other tests. the tests wouldn't pass otherwise